src/Adapter/FlysystemItems.php

```diff
@@ -0,0 +1,190 @@
+            throw new UnexpectedVersionNumber($id, $version, $highest);
+        }
+
+        if (1 === $version->toInt()) {
+            if (!$this->filesystem->deleteDir((string) $id)) {
+                throw new RuntimeException("Failed to remove item '{$id}'");
+            }
+
+            return;
+        }
+
+        try {
+            if (!$this->filesystem->delete("{$id}/{$version->toInt()}.xml")) {
+                throw new RuntimeException("Failed to remove item '{$id}' version {$version->toInt()}");
+            }
+        } catch (FileNotFoundException $e) {
+            return;
+        }
+    }
+
+    public function get(ItemId $id, ?ItemVersionNumber $version = null) : ItemVersion
+    {
+        if (!$this->filesystem->has((string) $id)) {
+            throw new ItemNotFound($id);
+        }
+
+        if (null === $version) {
+            $list = $this->filesystem->listContents((string) $id);
+            usort(
+                $list,
+                function (array $a, array $b) : int {
+                    return strnatcmp($b['filename'], $a['filename']);
+                }
+            );
+            $version = ItemVersionNumber::fromString($list[0]['filename']);
+        }
+
+        try {
+            $content = $this->filesystem->readStream("{$id}/{$version->toInt()}.xml");
+        } catch (FileNotFoundException $e) {
+            throw new VersionNotFound($id, $version, $e);
+        }
+
+        if (false === $content) {
+            throw new RuntimeException('Failed to read');
+        }
+
+        return new ItemVersion($id, $version, $content, hash_for_stream($content));
+    }
+
+    public function list(int $limit = 10, ?ItemId $startAt = null) : ItemListPage
+    {
+        $ids = array_map(
+            function (ItemVersion $item) : ItemId {
+                return $item->getId();
+            },
+            iterator_to_array($this->getIterator())
+        );
+
+        if (null !== $startAt) {
+            $key = array_search((string) $startAt, $ids);
+
+            if (false === $key) {
+                return new ItemListPage([], null);
+            } else {
+                $key = (int) $key;
+            }
+        }
+
+        $slice = array_slice($ids, $key ?? 0, $limit + 1);
+
+        if (count($slice) > $limit) {
+            $nextId = ItemId::fromString((string) array_pop($slice));
+        }
+
+        return new ItemListPage($slice, $nextId ?? null);
+    }
+
+    public function count() : int
+    {
+        return count(iterator_to_array($this->getIterator()));
+    }
+
+    public function getIterator() : Traversable
+    {
+        $items = array_filter(
+            $this->filesystem->listContents(),
+            function (array $item) : bool {
+                return 'dir' === $item['type'];
+            }
+        );
+
+        usort(
+            $items,
+            function (array $a, array $b) : int {
+                $now = time();
```

Comment:
The `$now` default can vary across calls from `usort` to the comparison function